common/txmgr/types/head_view.go

```diff
@@ -0,0 +1,30 @@
+package types
+
+import "github.com/ethereum/go-ethereum/common"
+
+// Provides a minimal access to a chain's head, as needed by the TxManager.
+// This is a generic interface whcih ALL chains will implement.
+type HeadView interface {
+	// The head's block number
+	BlockNumber() int64
+
+	// ChainLength returns the length of the chain followed by recursively looking up parents
+	ChainLength() uint32
+
+	// EarliestInChain recurses through parents until it finds the earliest one
+	EarliestInChain() HeadView
+
+	// The head's block hash
+	Hash() common.Hash
+
+	// The head's parent block
+	Parent() HeadView
+
+	// HashAtHeight returns the hash of the block at the given height, if it is in the chain.
+	// If not in chain, returns the zero hash
+	HashAtHeight(blockNum int64) common.Hash
+
+	// Returns the head in the chain's native type
+	// Chain specific code can retrieve the native type via this function.
+	GetNativeHead() interface{}
```

Comment:
i think using generics here might be better than `interface{}`

that way you only need to do the unwrapping in one place rather than multiple places?

Reply:
Thanks, done!